api/src/service/domain/organization/user_permission_grant.ts

```diff
@@ -0,0 +1,69 @@
+import isEqual = require("lodash.isequal");
+
+import Intent from "../../../authz/intents";
+import { Ctx } from "../../../lib/ctx";
+import * as Result from "../../../result";
+import { BusinessEvent } from "../business_event";
+import { InvalidCommand } from "../errors/invalid_command";
+import { NotAuthorized } from "../errors/not_authorized";
+import { NotFound } from "../errors/not_found";
+import { Identity } from "./identity";
+import { ServiceUser } from "./service_user";
+import * as Project from "../workflow/project";
+import * as UserRecord from "./user_record";
+import * as ProjectEventSourcing from "../workflow/project_eventsourcing";
+import * as UserEventSourcing from "./user_eventsourcing";
+import * as UserPermissionGranted from "./user_permission_granted";
+import { PreconditionError } from "../errors/precondition_error";
+
+interface Repository {
+  getUser(userId: UserRecord.Id): Promise<Result.Type<UserRecord.UserRecord>>;
```

Comment:
`getPermissions` instead?

Reply:
good idea! but how should we then validate the event? in [line 58](https://github.com/openkfw/TruBudget/pull/311/files/edca4ce6f95bdea6eb703af7a361ec0315dabd01#diff-59fa7b2a05c74650348ac63bc68e30feR58) we perform event sourcing on this user to check the validity.